tests/integration/thrift_tests.py

```diff
@@ -38,267 +42,386 @@
 )
 
 
-class UnexpectedException(Exception):
+cryptography_installed = True
+try:
+    import cryptography
+except:
+    cryptography_installed = False
+else:
+    del cryptography
+
+
+try:
+    from importlib import reload
+except ImportError:
     pass
 
 
-class TestHandler(TestService.Iface):
-    def example_simple(self, context):
-        return True
+def make_edge_context_factory():
+    mock_filewatcher = mock.Mock(spec=FileWatcher)
+    mock_filewatcher.get_data.return_value = {
+        "secrets": {
+            "secret/authentication/public-key": {
+                "type": "versioned",
+                "current": AUTH_TOKEN_PUBLIC_KEY,
+            },
+        },
+        "vault": {
+            "token": "test",
+            "url": "http://vault.example.com:8200/",
+        }
+    }
+    secrets = store.SecretsStore("/secrets")
+    secrets._filewatcher = mock_filewatcher
+    return EdgeRequestContextFactory(secrets)
+
+
+@contextlib.contextmanager
+def serve_thrift(handler, server_span_observer=None):
+    # create baseplate root
+    baseplate = Baseplate()
+    if server_span_observer:
+        class TestBaseplateObserver(BaseplateObserver):
+            def on_server_span_created(self, context, server_span):
+                server_span.register(server_span_observer)
+        baseplate.register(TestBaseplateObserver())
+
+    # set up the server's processor
+    logger = mock.Mock(spec=logging.Logger)
+    edge_context_factory = make_edge_context_factory()
+    processor = TestService.Processor(handler)
+    processor = baseplateify_processor(processor, logger, baseplate, edge_context_factory)
+
+    # bind a server socket on an available port
+    server_bind_endpoint = config.Endpoint("127.0.0.1:0")
+    listener = make_listener(server_bind_endpoint)
+    server = make_server({
+        "max_concurrency": "100",
+    }, listener, processor)
+
+    # figure out what port the server ended up on
+    server_address = listener.getsockname()
+    server_endpoint = config.EndpointConfiguration(
+        family=server_bind_endpoint.family, address=server_address)
+    server.endpoint = server_endpoint
+
+    # run the server until the user our caller with it
```

Comment:
typo?

Reply:
whoops yeah, also made me realize there's a missing try/finally around the `yield`. thanks!